app/models/katello/concerns/hostgroup_extensions.rb

```diff
@@ -12,6 +12,7 @@ module HostgroupExtensions
         belongs_to :lifecycle_environment, :inverse_of => :hostgroups, :class_name => "::Katello::KTEnvironment"
 
         validates_with Katello::Validators::ContentViewEnvironmentValidator
+        validates_with Katello::Validators::KickstartRepositoryValidator
```

Comment:
Can you ensure this validation happens only if KS repo id is set?

Reply:
@parthaa  Good call. Updated!